docs/design/environment/enviroment_management.md

```diff
@@ -0,0 +1,153 @@
+2. Only consider using Kusion in CLI format, the usage of service layer is not considered.
+
+## Proposal
+
+### Definition
+
+Environment is a logical concept representing a target that a stack will be deployed to, which contains a set of configurations, kubeconfig and provider authentication information that can be reused by multiple stacks. We recommend environments grouped by SDLC phases or cloud vendors. For example, it could be named as dev, staging, and prod or cloud vendors such as AWS, Azure, and Aliyun.
+
+At the product level, Kusion does not support multiple clouds or multiple regions in an environment. Although users can define an artful module which builds a schema into multiple resources in multiple clouds or regions. Kusion cannot forbid such modules, but doesn't recommend it and will not provide supportive technical means of multiple clouds or region in an environment. If a platform team needs to manage multiple clouds or regions, just create multiple environments. 
+
+Environment config are kept by yaml file when stored locally, and only includes a set of module inputs for now. But it may include other data, such as backend, which is not appropriate to set in the configuration file. There are two reasons: 1. backend info is a global config rather than environment config; 2. backend info includes sensitive data, such as database password.
+
+The kubeconfig and provider authentication information are kept by environment variables, cause the existence of sensitive data. For example, using AWS must set correct environment variables of `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY` and `AWS_REGION`.
+
+According to [Building the collaboration paradigm between App Developers and Platform Developers with Kusion.md](https://github.com/KusionStack/kusion/blob/main/docs/design/collaboration/Building%20the%20collaboration%20paradigm%20between%20App%20Developers%20and%20Platform%20Developers%20with%20Kusion.md), *a stack must be linked with one environment*. The stack name must be the same as the environment name it will deploy to.
+
+<br />![project-stack-env](project-stack-env.png)
+
+### Environment Config File Format
+
+The configuration of an environment should be maintained in a single yaml file, and the writing format and an example is shown as below.
+
+```yaml
+# Module input, with format standard：
+# # <ModuleName>:
+# #   default: # default configurations, applied to all projects
+# #     <ModuleInput_Field1>: <value1>
+# #     <ModuleInput_Field2>: <value2>
+# #     ...
+# #   <patcherName>: #patcher configurations, applied to the projects assigned in projectSelector
+# #     <ModuleInput_Field1>: <value1_Override>
+# #     ...
+# #     projectSelector:
+# #     - <projectName1>
+# #     ...
+module:
+  database:
+    default:
+      provider: aws
+      size: 20
+      instanceClass: db.t3.micro
+      securityIPs:
+      - 10.0.0.0/18
+    smallClass:
```

Comment:
Besides the override function, can this patcher add a new k-v pair into the default block?

Reply:
Yes, this is allowed.